### ISSUE TITLE
Uncaught Error: This TextEditor has been destroyed

### ISSUE THREAD

**Issue by fengzixu** [joefitzgerald/gofmt#12]
[Enter steps to reproduce below:]
1. ...
2. ...

**Atom Version**: 1.6.0
**System**: Mac OS X 10.11.3
**Thrown From**: [gofmt](https://github.com/joefitzgerald/gofmt) package, v1.1.3
### Stack Trace

Uncaught Error: This TextEditor has been destroyed

```
At /Applications/Atom 2.app/Contents/Resources/app.asar/src/pane.js:896

Error: This TextEditor has been destroyed
    at DisplayBuffer.module.exports.DisplayBuffer.screenPositionForBufferPosition (/Applications/Atom 2.app/Contents/Resources/app.asar/src/display-buffer.js:790:15)
    at DisplayBuffer.module.exports.DisplayBuffer.screenRangeForBufferRange (/Applications/Atom 2.app/Contents/Resources/app.asar/src/display-buffer.js:754:20)
    at TextEditorMarker.module.exports.TextEditorMarker.getScreenRange (/Applications/Atom 2.app/Contents/Resources/app.asar/src/text-editor-marker.js:141:33)
    at Selection.module.exports.Selection.getScreenRange (/Applications/Atom 2.app/Contents/Resources/app.asar/src/selection.js:78:26)
    at Selection.module.exports.Selection.finalize (/Applications/Atom 2.app/Contents/Resources/app.asar/src/selection.js:953:76)
    at Selection.module.exports.Selection.clear (/Applications/Atom 2.app/Contents/Resources/app.asar/src/selection.js:198:19)
    at Cursor.module.exports.Cursor.clearSelection (/Applications/Atom 2.app/Contents/Resources/app.asar/src/cursor.js:624:55)
    at Cursor.module.exports.Cursor.changePosition (/Applications/Atom 2.app/Contents/Resources/app.asar/src/cursor.js:670:12)
    at Cursor.module.exports.Cursor.setBufferPosition (/Applications/Atom 2.app/Contents/Resources/app.asar/src/cursor.js:80:19)
    at /Applications/Atom 2.app/Contents/Resources/app.asar/src/text-editor.js:1442:23
    at TextEditor.module.exports.TextEditor.moveCursors (/Applications/Atom 2.app/Contents/Resources/app.asar/src/text-editor.js:1679:9)
    at TextEditor.module.exports.TextEditor.setCursorBufferPosition (/Applications/Atom 2.app/Contents/Resources/app.asar/src/text-editor.js:1441:19)
    at Formatter.format (/Users/xuran/.atom/packages/gofmt/lib/formatter.js:306:16)
    at /Users/xuran/.atom/packages/gofmt/lib/formatter.js:99:14
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom 2.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:25:14)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom 2.app/Contents/Resources/app.asar/node_modules/event-kit/lib/emitter.js:125:28)
    at TextBuffer.module.exports.TextBuffer.saveAs (/Applications/Atom 2.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1051:20)
    at TextBuffer.module.exports.TextBuffer.save (/Applications/Atom 2.app/Contents/Resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1043:19)
    at TextEditor.module.exports.TextEditor.save (/Applications/Atom 2.app/Contents/Resources/app.asar/src/text-editor.js:629:26)
    at Pane.module.exports.Pane.saveItem (/Applications/Atom 2.app/Contents/Resources/app.asar/src/pane.js:613:18)
    at Pane.module.exports.Pane.saveActiveItem (/Applications/Atom 2.app/Contents/Resources/app.asar/src/pane.js:596:19)
    at Workspace.module.exports.Workspace.saveActivePaneItem (/Applications/Atom 2.app/Contents/Resources/app.asar/src/workspace.js:673:35)
    at atom-workspace.commandRegistry.add.core:save (/Applications/Atom 2.app/Contents/Resources/app.asar/src/register-default-commands.js:205:32)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom 2.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom 2.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom 2.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:570:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom 2.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:378:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeydown (/Applications/Atom 2.app/Contents/Resources/app.asar/src/window-event-handler.js:97:36)
    at HTMLDocument.<anonymous> (/Applications/Atom 2.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```
### Commands

```
     -1:02.8.0 core:paste (atom-text-editor.editor.is-focused)
     -0:58.6.0 core:copy (atom-text-editor.editor.is-focused)
     -0:55.8.0 core:paste (atom-text-editor.editor.is-focused)
 10x -0:55.3.0 core:save (atom-text-editor.editor.is-focused)
  2x -0:47.3.0 editor:newline (atom-text-editor.editor.is-focused)
 10x -0:40.4.0 core:save (atom-text-editor.editor.is-focused.autocomplete-active)
  2x -0:30.6.0 linter:next-error (atom-workspace.workspace.scrollbars-visible-when-scrolling.theme-seti-syntax.theme-seti-ui)
     -0:16.7.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:16.7.0 core:cancel (atom-text-editor.editor.is-focused)
  5x -0:08.9.0 core:save (atom-text-editor.editor.is-focused)
     -0:04.4.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:04.4.0 core:cancel (atom-text-editor.editor.is-focused)
  5x -0:03.4.0 core:save (atom-text-editor.editor.is-focused)
     -0:02.3.0 editor:consolidate-selections (atom-text-editor.editor.is-focused)
     -0:02.3.0 core:cancel (atom-text-editor.editor.is-focused)
  4x -0:01.3.0 core:save (atom-text-editor.editor.is-focused)
```
### Config

``` json
{
  "core": {
    "disabledPackages": [
      "linter-golint",
      "linter-golinter"
    ],
    "themes": [
      "seti-ui",
      "seti-syntax"
    ]
  },
  "gofmt": {
    "formatTool": "gofmt"
  }
}
```
### Installed Packages

``` coffee
# User
atom-beautify, v0.28.27
atom-html-preview, v0.1.19
autocomplete-go, v1.0.4
environment, v1.2.0
fonts, v1.2.0
go-config, v1.1.2
go-get, v1.0.1
go-plus, v4.0.1
gofmt, v1.1.3
gometalinter-linter, v1.0.6
gorename, v1.0.0
highlight-selected, v0.11.2
linter, v1.11.4
navigator-godef, v1.0.2
seti-syntax, v0.4.2
seti-ui, v0.9.2
tester-go, v1.0.2

# Dev
No dev packages
```

**Comment by fengzixu** [comment]
l

**Comment by joefitzgerald** [comment]
This is fixed in `v1.1.4`.
